### PR TITLE
Eui 1994 action and labels

### PR DESCRIPTION
**Pull request by kpokala**
Resolves # . (This is applicable only if this pull request relates to a GitHub issue, delete the line otherwise)

Notes:
*
*
*

### CODE REVIEW

File: projects/exui-common-lib/src/lib/services/case-sharing-state/case-sharing-state.service.ts [hunk 1]
```diff
@@ -20,59 +20,115 @@ export class CaseSharingStateService {
...
+        sharedCase.pendingShares = [];
+      }
+      const newPendingShares = sharedCase.pendingShares.slice();
+      if (!sharedCase.pendingUnshares) {
+        sharedCase.pendingUnshares = [];
+      }
+      const newPendingUnshares = sharedCase.pendingUnshares.slice();
+      if (!this.userHasAccess(sharedCase, user)) {
+        if (!newPendingShares.some(u => u.email === user.email)) {
+          newPendingShares.push(user);
```
Comment (udaydenduluri33) [inline projects/exui-common-lib/src/lib/services/case-sharing-state/case-sharing-state.service.ts:50]: Please use spread operator

File: projects/exui-common-lib/src/lib/services/case-sharing-state/case-sharing-state.service.ts [hunk 2]
```diff
@@ -20,59 +20,115 @@ export class CaseSharingStateService {
...
-        // done dealing with this case, so exit early
-        this.subject.next(this.caseState);
-        return;
       }
+      const newSharedCase = {
+        ...sharedCase,
+        pendingUnshares: newPendingUnshares,
+        pendingShares: newPendingShares
+      };
+      newSharedCases.push(newSharedCase);
```
Comment (udaydenduluri33) [inline projects/exui-common-lib/src/lib/services/case-sharing-state/case-sharing-state.service.ts:64]: Please use spread operator

File: projects/exui-common-lib/src/lib/services/case-sharing-state/case-sharing-state.service.ts [hunk 3]
```diff
@@ -20,59 +20,115 @@ export class CaseSharingStateService {
...
   constructor() { }
 
-  public setCases(orgId: string, cases: SharedCase[]): void {
-    // TODO: check organisation policy against orgId
-    parseInt(orgId, 10); // do something with orgId for now to stop linting errors
+  public setCases(cases: SharedCase[]): void {
     this.caseState = [];
     cases.forEach(c => {
-      this.caseState.push({ caseId: c.caseId, caseTitle: c.caseTitle, roles: c.roles, sharedWith: c.sharedWith, pendingShares: c.pendingShares, pendingUnshares: c.pendingUnshares });
+      this.caseState.push({ caseId: c.caseId, caseTitle: c.caseTitle, roles: c.roles,
```
Comment (udaydenduluri33) [inline projects/exui-common-lib/src/lib/services/case-sharing-state/case-sharing-state.service.ts:26]: Please use spread operator

File: projects/exui-common-lib/src/lib/components/selected-case/selected-case.component.spec.ts [hunk 4]
```diff
@@ -43,4 +51,184 @@ describe('SelectedCaseComponent', () => {
...
+      idamId: 'pus111111',
+      firstName: 'JamesPUS',
+      lastName: 'PriestPUS',
+      email: 'jamespus.priestpus@test.com'
+    };
+
+    expect(component.isToBeRemoved('C111111', user)).toEqual(true);
+  });
+
+
```
Comment (sreekanthpuligadda) [inline projects/exui-common-lib/src/lib/components/selected-case/selected-case.component.spec.ts:228]: Please add tests to verify html template and events from template,

1. if users listed and correct number of users displayed
2. If Remove label displayed
3. If Cancel label displayed
4. If to be removed label displayed
5. If to be added label is displayed
6. If onRemove triggered on clicking Remove link
7. If onCancel triggered on clicking cancel link

File: projects/exui-common-lib/src/lib/services/case-sharing-state/case-sharing-state.service.spec.ts [hunk 5]
```diff
@@ -1,12 +1,101 @@
...
+  });
+
+  it('should set cases', () => {
+    sharedCases  = [{
+      caseId: '9417373995765133',
+      sharedWith: [],
+      caseTitle: 'Sam Green Vs Williams Lee'
+    }];
+    service.setCases(sharedCases);
+    expect(service).toBeTruthy();
```
Comment (sreekanthpuligadda) [inline projects/exui-common-lib/src/lib/services/case-sharing-state/case-sharing-state.service.spec.ts:28]: Validation missing. Validate if case added to service.

File: projects/exui-common-lib/src/lib/services/case-sharing-state/case-sharing-state.service.spec.ts [hunk 6]
```diff
@@ -1,12 +1,101 @@
...
+    }];
+    let user: UserDetails = {
+      idamId : 'pus111111',
+      firstName: 'JamesPUS',
+      lastName: 'PriestPUS',
+      email: 'jamespus.priestpus@test.com'
+    };
+    service.setCases(sharedCases);
+    service.requestUnshare('9417373995765133', user);
+    expect(service).toBeTruthy();
```
Comment (sreekanthpuligadda) [inline projects/exui-common-lib/src/lib/services/case-sharing-state/case-sharing-state.service.spec.ts:63]: Validation missing. Validate if user not in pendingUnshare list.

File: projects/exui-common-lib/src/lib/services/case-sharing-state/case-sharing-state.service.spec.ts [hunk 7]
```diff
@@ -1,12 +1,101 @@
...
+    }];
+    let user: UserDetails = {
+      idamId : 'pus111111',
+      firstName: 'JamesPUS',
+      lastName: 'PriestPUS',
+      email: 'jamespus.priestpus@test.com'
+    };
+    service.setCases(sharedCases);
+    service.requestCancel('9417373995765133', user);
+    expect(service).toBeTruthy();
```
Comment (sreekanthpuligadda) [inline projects/exui-common-lib/src/lib/services/case-sharing-state/case-sharing-state.service.spec.ts:86]: Validation missing. Validate user is not in pendingShares and pendingUnshrares list.
May need to add new test for PendingShares scenario.

File: projects/exui-common-lib/src/lib/components/share-case/share-case.component.spec.ts [hunk 8]
```diff
@@ -41,8 +47,122 @@ describe('ShareCaseComponent', () => {
...
+      email: 'jamespus.priestpus@test.com'
+    };
+
+    component.onSelectedUser(user);
+    component.shareCases$ = of(sharedCases);
+    component.addUser();
+    fixture.detectChanges();
+    expect(component.isDisabledContinue()).toBeFalsy();
+  });
+
```
Comment (sreekanthpuligadda) [inline projects/exui-common-lib/src/lib/components/share-case/share-case.component.spec.ts:164]: Add tests for addUser method and html trigger for Add User click.

File: projects/exui-common-lib/src/lib/services/case-sharing-state/case-sharing-state.service.spec.ts (hunk 5 above)
Comment (kpokala) [reply projects/exui-common-lib/src/lib/services/case-sharing-state/case-sharing-state.service.spec.ts]: modified the change by size.

File: projects/exui-common-lib/src/lib/services/case-sharing-state/case-sharing-state.service.spec.ts (hunk 7 above)
Comment (kpokala) [reply projects/exui-common-lib/src/lib/services/case-sharing-state/case-sharing-state.service.spec.ts]: modified the chnage by checking size

**Pull request merged**